scrapers/covid-19/govScrapers/getMexico.js

```diff
@@ -78,25 +98,59 @@ const getState = (res) => {
 	return stateValues;
 };
 
+/**
+ * Gets national Active, Negative, Suspect, and Recovered counts
+ * @param {string} res	The response body string to extract data from
+ * @returns {Object}	{ active, negative, suspect, recovered }
+ */
+const getActNegSusRec = (res) => ({
+	active: getInnerHTML(res, 'gsActDIV'),
+	negative: getInnerHTML(res, 'gsNegDIV'),
+	suspect: getInnerHTML(res, 'gsSosDIV'),
+	recovered: getInnerHTML(res, 'gsRecDIV')
+});
+
 const mexicoData = async () => {
+	const SOURCE_URL = 'https://coronavirus.gob.mx/datos/Overview/info/getInfo.php';
+
 	try {
-		const nationalCases = getNational((await axios.post('https://coronavirus.gob.mx/datos/Overview/info/getInfo.php', formData.nationalCases)).data);
-		const nationalDeaths = getNational((await axios.post('https://coronavirus.gob.mx/datos/Overview/info/getInfo.php', formData.nationalDeaths)).data);
-		const stateCases = getState((await axios.post('https://coronavirus.gob.mx/datos/Overview/info/getInfo.php', formData.stateCases)).data);
-		const stateDeaths = getState((await axios.post('https://coronavirus.gob.mx/datos/Overview/info/getInfo.php', formData.stateDeaths)).data);
+		const nationalCases = getNational((await axios.post(SOURCE_URL, formData.nationalCases)).data);
+		const nationalDeaths = getNational((await axios.post(SOURCE_URL, formData.nationalDeaths)).data);
```

Comment:
since you added a return value from `getNational`, keep in mind this could be set to null is something errors. Maybe a more appropriate error value is the empty list? I only say this because if people are expecting a list you don't want to implode their program by having them check `null.length` or something

Reply:
you're correct, `getNational` relies on `filterByDate`.

but what would actually happen if `filterByDate` fails is, it would still return an empty array, because Array.filter returns [] if the predicate is never met. this function would only reach the catch if `data` is undefined, which would again bubble up to `mexicoData` where the post request was made.

then the error would bubble up to `mexicoData`, where the catch would again return null. this brings up to `getGovData`. 

see my message on discord about redis